### ISSUE TITLE
Dashboard redesign

### ISSUE THREAD

**Issue by novabyte** [heroiclabs/nakama#116]
The dashboard which comes with Nakama server was re-enabled in the 1.0.2 release ahead of a redesign. This issue tracks design requirements and requests from the community for what they'd like to see in the reworked dashboard.

**Comment by Haones** [comment]
The most important thing, which dashboard will do is retrieving information from database through server requests. In which case else dashboard will be useful? For getting main information about configuration, managing access to server from client (I don't think you can update everything on fly, sometimes you will need to tell users that server will be updated and you can set server status from dashboard) checking server health and etc.

But, returning to the feature of reading database, of course, every project will have different data, which is stored in database and you cannot make all possible database requests by default for dashboard. In this case, it will be perfect to have possibility to add/make custom modules (in Go language in our case), like Django web-framework to work with data (read, add new records and etc) from cockroach database or collections. 

In other words, may be we can use some existing web-frameworks to make data managing from dashboard much simple than fetching records from CLI? I think, this is would be main advantage of dashboard.

**Comment by milox** [comment]
Definitely the ability to edit the all data, without needing to do it via code. See all the players and their data, and be able to edit it. Easy way to set remote config.

See Playfab cloud service how a decent dashboard should look like, although theirs is too complicated imo, better to keep it simple.

**Comment by Haones** [comment]
> although theirs is too complicated imo
Exactly. Because they trying to predict every case in every game project. But most of modules you will never use. That's why custom modules is better solution in this case with open-source project.

**Comment by mofirouz** [comment]
There are a few different things mentioned in here that touch upon various aspects of a dashboard. Typically the point of an embedded dashboard is to allow operators to inspect the running state of the application, make configuration changes (and potentially apply them on the fly), and get a sense of what the system performance is like.

An example is RabbitMQ's embedded dashboard that exposes metric information, lets you configure exchanges and queues and other config options like setting policies and so on. This is similar to CockroachDB's embedded dashboard that exposes only metrics.

> Definitely the ability to edit the all data, without needing to do it via code. See all the players and their data, and be able to edit it. Easy way to set remote config.

This is an entirely different dashboard domain. User management, views for supporting staff, different user logins with different access control levels, setting up live events and remove configuration will not be part of an embedded dashboard. What you are looking for is a live-ops dashboard which we are providing as a different product entirely and won't be part of Nakama.

Having said all of this, we are listening to the community and we will add read-only views to fetch some basic information from the database. This is purely to help developers when they are building their project with Nakama.

**Comment by milox** [comment]
Ah I see.  Will the live ops dash be available as a standalone addon for nakama or just on your cloud or..?

**Comment by mofirouz** [comment]
@milox The LiveOps Dashboard is a standalone product that works with Nakama, and has a modular layout as @haones mentioned so that custom views can be added based on needs. It works with our cloud option and can also be deployed on your own.

**Comment by bluescrn** [comment]
After getting quite excited about giving Nakama a try, I was a rather disappointed to see that the dashboard for the open source server was just an 'it's up' screen, without even a log viewer or configuration options, let alone basic data viewing/editing.  And it seems that there's currently a lack of GUI tools for working directly with CockroachDB, too.

What you've branded as the 'LiveOps Dashboard' looks like an important part of the development process too. But with no pricing information on your website, it seems like a case of 'if you have to ask, you can't afford it' (to a solo indie developer, at least)?

**Comment by ulasbilgen** [comment]
@bluescrn we’re using Nakama as an indie studio and it’s installed on our free Azure subscription, which we got through BizSpark program.
Anyhow, we also needed to able to change the data in the DB, so we just put together a very simple HTML, node.js + Lu’s module, if you want I can share that with you. 
It’s quite basic but it does do the job :)

**Comment by milox** [comment]
@ulasbilgen Please do share with everyone

**Comment by bluescrn** [comment]
@ulasbilgen Right now, I'm mostly just weighing up the options out there for a back end for a small project, but yes, it looks like building some custom tools would be on the list of things to do if I went with Nakama.

(I guess theoretically, there could be an open-source effort to build an enhanced dashboard - but it seems wrong to encourage this if it's going to undermine the business model of Heroic Labs)

**Comment by Ceiyne** [comment]
I've only just started to use Nakama recently, so my feedback may not be as fully-formed as someone who's been using it for a while or in a production app.  Thinking about what I'd expect to need in the future for my project, I would want to be able to do things like:

- Run SQL and other database commands, and view the results/output
- Perform admin tasks like locking/unlocking user accounts or resetting user passwords
- Set server parameters/configuration
- Edit, register, enable, and disable server-side scripting (Lua)

**Comment by novabyte** [comment]
The initial release of the new Developer Console was added in the [2.5.0](https://github.com/heroiclabs/nakama/releases/tag/v2.5.0) release.
